src/Workleap.Extensions.Mongo/Indexing/IndexDeleter.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Workleap.Extensions.Mongo.Indexing;
+
+/// <summary>
+/// Delete indexes for a particular collection by comparing existing indexes in the database
+/// with the expected indexes declared in the code.
+/// </summary>
+internal sealed class IndexDeleter
+{
+    private readonly IMongoDatabase _database;
+    private readonly ILogger _logger;
+    
+    public IndexDeleter(IMongoDatabase database, ILogger<IndexDeleter> logger)
+    {
+        this._database = database;
+        this._logger = logger;
+    }
+    
+    public async Task ProcessAsync(Dictionary<string, IList<UniqueIndexName>> expectedIndexes, CancellationToken cancellationToken)
+    {
+        foreach (var expectedIndexForCollection in expectedIndexes)
+        {
+            var existingIndexesForCollection = await this.GetExistingIndexes(expectedIndexForCollection.Key).ConfigureAwait(false);
```

Comment:
Could we somehow leverage what we do in PopulateExistingIndexes and load all indexes only once instead of on every collection?

Reply:
Not sure since we know the indexes will be outdated since we populate the indexes before creating the indexes